packages/@ember/debug/tests/main_test.js

```diff
@@ -273,6 +328,77 @@ moduleFor(
       );
     }
 
+    ['@test deprecate without options.for triggers an assertion'](assert) {
+      assert.expect(1);
+
+      assert.throws(
+        () => deprecate('message1', false, { id: 'test', until: 'forever' }),
+        new RegExp(missingOptionsForDeprecation('test')),
+        'proper assertion is triggered when options.for is missing'
+      );
+    }
+
+    ['@test deprecate without options.for only triggers once per id'](assert) {
+      ENV.RAISE_ON_DEPRECATION = false;
+      let messages = [];
+      registerHandler(function (message, options, next) {
+        if (options.id === 'ember-source.deprecation-without-for') {
+          messages.push(message);
+        }
+        next(...arguments);
+      });
+
+      deprecate('message1', false, { id: 'test', until: 'forever' });
+      deprecate('message2', false, { id: 'test', until: 'forever' });
+      deprecate('message3', false, { id: 'another', until: 'forever' });
+
+      assert.equal(messages.length, 2, 'correct number of deprecations');
+      assert.equal(messages[0], missingOptionsForDeprecation('test'), 'first message is correct');
+
+      assert.equal(
+        messages[1],
+        missingOptionsForDeprecation('another'),
+        'second message is correct'
+      );
+    }
+
+    ['@test deprecate without options.since triggers an assertion'](assert) {
+      assert.expect(1);
+
+      assert.throws(
+        () =>
+          deprecate('foo', false, {
+            id: 'test',
+            until: 'forever',
+            for: 'me',
+          }),
+        new RegExp(missingOptionsSinceDeprecation('test')),
+        'proper assertion is triggered when options.since is missing'
+      );
+    }
```

Comment:
Issues a deprecation, not an assertion.

```suggestion
    ['@test deprecate without options.since triggers a deprecation'](assert) {
      assert.expect(1);

      expectDeprecation(
        () =>
          deprecate('foo', false, {
            id: 'test',
            until: 'forever',
            for: 'me',
          }),
        new RegExp(missingOptionsSinceDeprecation('test')),
        'deprecation is triggered when options.since is missing'
      );
    }
```